### DISCUSSION TITLE
My email is hacked, I cannot recover , how can I change my GitHub email to my new email?

### DISCUSSION THREAD

**Discussion by im-Nitesh** [community/community#139508] [Code Security] [answered]
### Select Topic Area

Question

### Body

Hi ,

My gmail is hacked, and I cannot recover , how can I change my GitHub email to my new email?

thanks in advance

**Comment by im-shubh** [comment, marked as answer]
Hey @im-Nitesh  

If your Gmail is hacked and you're unable to recover it, here's how you can change your GitHub email to a new one:

**1. Log into GitHub**  using your existing credentials.

**2. Add a new email:**
->Go to **[Settings > Emails.](https://github.com/settings/emails)** 
->Under **Add email address,** enter your new email and click Add.

**3. Verify your new email:**
GitHub will send a verification link to your new email. Click on the link to verify it.

**4. Set your new email as primary:**
->After verifying the new email, go back to **[Settings > Emails](https://github.com/settings/emails).**
->Next to your new email, click **Make primary**.

**5. Remove your old email:**
Once your new email is set as primary, you can remove the old (hacked) email from your account.

**6. Enable 2FA (Two-Factor Authentication):**
For added security, consider enabling two-factor authentication in **Settings > Account security**.

Let me know if you need further assistance.

Thanks!

**Reply by amandacepeda333-cell** [reply]
They've hacked my account. They put in a new email address as Amanda cepeda333 @Gmail.com which is not my email address. They have taken all of my information. I would like to take their their email off of my accounts and report them. They are stealing money from me stealing my identity. These are really bad people and I've turned them into the police department. They have all of their information and they will be found and caught

**Reply by amandacepeda333-cell** [reply]
Thank you for all the information and how to change it. it's really hard and really messed up that they've already gotten all of my information and have taken all of it and started putting credit cards in my name and every other day. I'm getting messages saying that they are taking money out of my PayPal and it's all because that I had some information on this site that I did not know that was able to be given to anybody. I seen the guy asked the person on here about my information and how silly it was that they needed it and they never questioned why they would need that personal information to access my accounts. But they just gave it to them. And now I have to report everything to the police department and I have to pray to the Lord that I don't go into debt. It's just wrong that you can get information that quickly from something that I thought was so secure, but I guess that's how naive I am. I really appreciate your help. They've already taken so much. I just want my site shut down now. I can't even get a place to live because I have this person acting like me and taking everything. I just want everybody to be careful cuz it's so easy to happen to you

**Comment by arslan578** [comment]
To change your GitHub email if you no longer have access to your Gmail account, follow these steps:

1. **Log into GitHub** using your username and password (if you're already logged in, skip this step).

2. Go to **Settings**:
   - Click on your profile picture in the upper-right corner.
   - Select **Settings** from the dropdown menu.

3. **Update your email**:
   - In the left sidebar, click on **Emails**.
   - Add your **new email address** by entering it in the "Add email address" field.
   - Click **Add**.

4. **Verify your new email**:
   - GitHub will send a verification email to your new address.
   - Open the email and follow the verification link.

5. **Set the new email as primary**:
   - Once verified, go back to **Emails** in the settings.
   - Set the new email as your **primary email** by clicking the "Set as primary" button next to your new email.

6. **Remove the old email**:
   - After updating, you can safely remove the hacked Gmail by clicking the **Delete** icon next to it.

This will update your GitHub account with your new email address. ✔️

**Reply by amandacepeda333-cell** [reply]
You could help me if I give you my new email address?

**Reply by amandacepeda333-cell** [reply]
Sorry I read that wrong but I understand what you're saying. I can update it

**Reply by Anandakeylines** [reply]
after clicking add it send otp to my old email but I dont have access to old email

**Comment by Orangnoorshamsi** [comment]
I don't have access to my original and old Gmail because I forgot my password and the verification steps. Please help me activate my old Gmail.

**Comment by Orangnoorshamsi** [comment]
I don't have access to my original and old Gmail because I forgot my password and the verification steps. Please help me activate my old Gmail.

**Comment by amandacepeda333-cell** [comment]
I just want my life back and I want them to stop taking from me. I don't have anything to give. I'm disabled and homeless and they're stealing everything I have with a child that has five mental illnesses. They have not a care in the world but themselves. These are really terrible people and they need to be stopped. Please help me stop them from doing any more harm. I already got pictures of them and also names but I need you guys to shut down the site that is allowing them to get more and more information

**Reply by amandacepeda333-cell** [reply]
And please just be careful for this person that is acting as me. Amandacepeda333@gmail.com

**Comment by amandacepeda333-cell** [comment]
Thank you all for helping me and understanding. I really appreciate each and every one of you and I hope you all have a great day and be blessed in everything you do

**Comment by naqvi110k** [comment]
I went through this nightmare last year when my old university email got deactivated. The standard "add a new email" flow everyone keeps posting **does not work** if you can't get the verification code sent to your hacked/dead email. I see a few people in the thread already hitting that wall.

**The hard truth:** GitHub's automated system is designed to protect accounts from hijacking. If you can't prove ownership of the old email, the self-service flow dead-ends. You need a human.

**Here's the actual path that worked for me and two of my coworkers:**

1. **Go to GitHub Support** (support.github.com) – not the community forum. The community is great but no one here can reset your account.

2. **Select "Account" → "I can't sign in" → "I don't have access to my email"** (paraphrasing, but you'll see the options).

3. **In the ticket, include EVERYTHING you can prove:**
   - Your GitHub username
   - Any previous emails associated with the account (even if hacked)
   - Approximate account creation date
   - Any known repository names, organization memberships, or past contribution dates
   - **Crucially:** Proof that you can still *log in* to GitHub (if you're still signed in on a device, mention that). Being logged in helps prove you have the password, just not the email.

4. **If you have any payment method on file** (GitHub Pro, Copilot, etc.), include the last 4 digits of the card or any billing email receipts. This is the single fastest way to verify ownership.

**What to expect:** Support will take 3-7 days. They will likely ask you to verify via a secondary method (like answering security questions about your repos or providing a screenshot of your logged-in session). It's slow, but they will eventually help you add a new email without needing the old one.

**One warning:** If you cannot log in *at all* (lost password AND hacked email), you're in a much harder spot. In that case, support will ask for things like old commit hashes, SSH key fingerprints, or a notarized identity verification. It's possible but painful.

Hang in there. This is fixable, but you have to go through the official support ticket system, not the community workarounds.
